File: 80. Remove Duplicates from Sorted Array II.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int len = nums.size();
        if(len == 0)	return 0;
        int last(nums[0]), cnt = 1, i = 1, j = 1;
        while(i < len && j < len){
        	if(nums[j] != last){
        		last = nums[j];
        		cnt = 1;
        		nums[i++] = nums[j++];
        	}
        	else{
        		if(cnt < 2){
        			++cnt;
	        		nums[i++] = nums[j++];
        		}
        		else ++j;
        	}
        }
        return i;
    }
};

File: 80. Remove Duplicates from Sorted Array II.py
class Solution(object):

    def removeDuplicates(self, nums):

        """

        :type nums: List[int]

        :rtype: int

        """

        cnt = 0

        dpl = 0

        current = -sys.maxsize

        for i in range(len(nums)):

            if nums[i]>current:

                nums[cnt] = nums[i]

                current = nums[cnt]

                cnt += 1

                dpl = 1

            elif nums[i]==current and dpl==1:

                nums[cnt] = current

                dpl += 1

                cnt += 1

        return cnt
